src/plugins/intel_gpu/src/plugin/program.cpp

```diff
@@ -482,6 +483,24 @@ void Program::InitProfileInfo(const std::string& layerName,
     perfEntry.parentPrimitive = parentId;
 }
 
+cldnn::memory_ptr Program::GetVariableMemory(const std::string& variable_id, const cldnn::layout& layout) {
+    cldnn::memory_ptr memory;
+
+    const auto var = m_variables.find(variable_id);
+    if (var == m_variables.end()) {
+        memory = m_engine->allocate_memory(layout);
+        m_variables[variable_id] = cldnn::network::variable{memory};
+    } else {
+        memory = var->second.memory;
```

Comment:
How about checking "is_set" at this time? If is_set is false, the "memory" would be not valid, right?